lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -307,7 +307,8 @@ def get(self, request, *args, **kwargs):  # pylint: disable=too-many-statements
         #
         # The long term goal is to remove the Course Blocks API call entirely,
         # so this is a tiny first step in that migration.
-        if course_blocks:
+        user_is_audit = getattr(enrollment, 'mode', '') == 'audit'
```

Comment:
I'd recommend you use the following for great modularity.

- [CourseEnrollment.enrollment_mode_for_user](https://github.com/edx/edx-platform/blob/c523a5d9dde32f48c3b40e2eacfc652a7935ee25/common/djangoapps/student/models/course_enrollment.py#L941) to get the learner's enrollment mode
- UPSELL_TO_VERIFIED_MODES variable instead of hard-coding `audit`; you may consider adding a class method like [is_mode_upgradeable](https://github.com/edx/edx-platform/blob/release-ulmo/common/djangoapps/course_modes/models.py#L645) to make this code more modular